### PR TITLE
BigQuery partitioning

### PR DESCRIPTION
**Pull request by jskeet**
Best to review one commit at a time; there are a couple of naming questions in the first commit.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateTableOptions.cs [hunk 1]
```diff
@@ -46,15 +46,22 @@ public sealed class CreateTableOptions
...
 
         /// <summary>
         /// The time partition type for the table, if the table should be created with
         /// time-based partitioning.
         /// </summary>
+        [Obsolete("This property will be removed before final release. Please use TimePartition to create a TimePartitioning and set CreateTableOptions.TimePartitioning.")]
         public TimePartitionType? TimePartitionType { get; set; }
 
         /// <summary>
+        /// The time partitioning for the table. For a .NET-friendly
```
Comment (evildour) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateTableOptions.cs]: Looks like this comment wasn't finished.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TimePartition.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    /// Convenience methods for creating <see cref="TimePartitioning"/>.
+    /// </summary>
+    public static class TimePartition
+    {
+        /// <summary>
+        /// Returns a daily partitioning scheme with an optional per-partition expiration.
+        /// </summary>
+        /// <param name="expiration">The expiration duration for each partition, if any.</param>
+        /// <returns>A <see cref="TimePartitioning"/> representing a daily partitioning.</returns>
+        public static TimePartitioning CreateDailyPartitioning(TimeSpan? expiration)
```
Comment (evildour) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TimePartition.cs]: Perhaps default this parameter to null?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/CreateTableOptionsTest.cs [hunk 3]
```diff
@@ -45,12 +44,28 @@ public void PropertiesSetOnRequest()
             Assert.Same(options.ExternalDataConfiguration, table.ExternalDataConfiguration);
         }
 
+#pragma warning disable CS0618 // Type or member is obsolete
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/CreateTableOptionsTest.cs]: It's on line 99, after all the tests that deal with this.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateTableOptions.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateTableOptions.cs]: Fixed, and added something similar to the other options.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TimePartition.cs (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TimePartition.cs]: I was wondering about that. I think I'd rather not, in this case, as users should probably at least consider an expiry.

**Comment by jskeet** [comment]
(Will merge when CI has tested again.)

**Pull request merged**